### PR TITLE
Lock Button

### PR DESCRIPTION
**Pull request by ZackKuptsow**
Lock button toggle that changes url. 
Content includes locked option now. 
Adds pin to url if locked equals true.

### CODE REVIEW

**Comment by muke1908** [comment]
Hi @ZackKuptsow, thank you for the PR. Can you please fix the build, it's failing?
```
./src/pages/messaging/helpers/index.js
  Line 31:9:  'lockedStatus_storage' is assigned a value but never used  no-unused-vars

./src/components/LockToggle/index.js
  Line 6:10:   'LockContext' is defined but never used        no-unused-vars
  Line 19:10:  'darkMode' is assigned a value but never used  no-unused-vars
  Line 31:20:  Expected '===' and instead saw '=='            eqeqeq

./src/index.js
  Line 9:50:  'Link' is defined but never used      no-unused-vars
  Line 9:56:  'Redirect' is defined but never used  no-unused-vars
  ```

**Comment by ZackKuptsow** [comment]
@muke1908 removed the unnecessary code and files to pass the build test

File: client/package.json [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
     "@testing-library/user-event": "^7.2.1",
+    "bluebird": "^3.7.2",
     "pubnub": "^4.28.2",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-icons": "^3.10.0",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "3.4.1"
+    "react-scripts": "^3.4.4",
+    "website-scraper": "^4.2.3"
```
Comment (muke1908) [inline client/package.json:16]: looks like `website-scraper` and `bluebird` are unnecessary and never used

File: client/src/LockContext.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React, { useState, createContext } from 'react';
+import storage from './utils/storage.js';
+
+export const LockContext = createContext();
+
+let defaultLockMode = storage.get('locked');
+defaultLockMode = defaultLockMode === null ? false : defaultLockMode;
```
Comment (muke1908) [inline client/src/LockContext.js:7]:
```suggestion
const defaultLockMode = !!storage.get('locked');
```

File: client/src/components/LinkDisplay/index.js [hunk 3]
```diff
@@ -62,7 +64,7 @@ const LinkDisplay = ({ content }) => {
       >
         {/* todo: this needs to be changed - make an POST endpoint to fetch the PIN of an url */}
         <a
-          href={`${content.absoluteLink}?pin=${content.pin}`}
+          href={`${content.absoluteLink}?pin=${content.pin}&?locked=${content.locked}`}
```
Comment (muke1908) [inline client/src/components/LinkDisplay/index.js:67]: Why do you need to pass `locked` in link?

File: client/src/pages/chatlink/captcha.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+export const getCaptchaInstance = async () => {
+  if (window.grecaptcha) {
+    return window.grecaptcha;
+  } else {
+    await loadJS('https://www.google.com/recaptcha/api.js?render=explicit', document);
+    return window.grecaptcha;
+  }
+};
```
Comment (muke1908) [inline client/src/pages/chatlink/captcha.js:24]: This file is not used anywhere

File: client/src/components/LockToggle/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    storage.set('locked', !lockMode);
+    setLockMode(!lockMode);
+    const url = new URLSearchParams(window.location.search);
+    window.history.replaceState(
+      { locked: lockMode },
+      '',
+      `?pin=${url.get('pin')}&?locked=${!lockMode}` +
+        (!lockMode === true ? `&?extrapin=${extraPin(8)}` : '')
+    );
+  };
```
Comment (muke1908) [inline client/src/components/LockToggle/index.js:29]:
```suggestion
const toggleLock = () => {
    storage.set('locked', !lockMode);
    setLockMode(!lockMode);
}
```
This implementation needs to go to the backend. For now, let's leave it as suggested.

**Review by muke1908 — CHANGES_REQUESTED** [review]
As I understood you tried to work on issue #170. 
As this PR is partial implementation, I would like to merge it keeping the origin issue open. 
Please look at my comments.

**Pull request closed without merging**